caterva2/hdf5.py

```diff
@@ -67,12 +78,47 @@ def b2args_from_h5dset(h5_dset: h5py.Dataset) -> Mapping[str, object]:
     return b2_args
 
 
+def b2attrs_from_h5dset(
+        h5_dset: h5py.Dataset,
+        attr_ok: Callable[[h5py.Dataset, str], None] = None,
+        attr_err: Callable[[h5py.Dataset, str, Exception], None] = None) -> (
+            Mapping[str, object]):
+    """Get msgpack-encoded attributes from the given HDF5 dataset.
+
+    If given, call `attr_ok` or `attr_err` on attribute translation success or
+    error, respectively.
+    """
+    b2_attrs = {}
+    for (aname, avalue) in h5_dset.attrs.items():
+        try:
+            # This small workaround avoids Blosc2's strict type packing,
+            # so we can handle value subclasses like `numpy.bytes_`
+            # (e.g. for Fortran-style string attributes added by PyTables).
+            pvalue = msgpack.packb(avalue, default=blosc2_ext.encode_tuple)
```

Comment:
This can handle most of the Python values, but NumPy objects are not supported.  May be using https://github.com/lebedov/msgpack-numpy could be a solution here?

Reply:
Can you do a test with https://github.com/lebedov/msgpack-numpy and see if that would work for Caterva2?